current.go

```diff
@@ -0,0 +1,79 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/finalizerskeptcontext"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	clusterID, err := clusterID(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for services in the Kubernetes API")
+
+	var services []*corev1.Service
+
+	namespace := clusterNamespace(clusterID)
+	serviceNames := []string{
+		serviceNameMaster,
+	}
+	if isKVM(obj) {
+		serviceNames = append(serviceNames, serviceNameWorker)
+	}
```

Comment:
This has to be configurable so the provider operators can set their own key functions.